### PR TITLE
test(NODE-5254): use node 20 in actions

### PR DESCRIPTION
**Pull request by durran**
### Description

Use Node 20 in CI.

#### What is changing?
- adds Node 20.

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

NODE-5254

### Double check the following

- [x] Ran `npm run format:js && npm run format:rs` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -285,11 +285,17 @@ jobs:
           key: npm-cache-test-linux-aarch64-gnu-${{ matrix.node }}-${{ hashFiles('package-lock.json') }}
       - name: Install dependencies
         run: npm clean-install --ignore-scripts
+      - name: Set up QEMU
```
Comment (durran) [inline .github/workflows/CI.yml:288]: Got around the missing docker image in napi-rs for Node 20 by using QEMU with a standard Node 20 image.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -148,10 +148,10 @@ jobs:
           - host: windows-latest
             target: x86_64-pc-windows-msvc
         node:
-          - '12'
```
Comment (dariakp) [inline .github/workflows/CI.yml:151]: why did we want to drop support for Node 12 here?

File: .github/workflows/CI.yml (hunk 1 above)
Comment (dariakp) [reply .github/workflows/CI.yml:288]: is this setup incompatible with Node 12?

File: .github/workflows/CI.yml (hunk 2 above)
Comment (durran) [reply .github/workflows/CI.yml]: I've brought back Node 20 and updated the readme.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (durran) [reply .github/workflows/CI.yml:292]: Nope. I've brought it back now. I'll file a follow up ticket to remove Node 12 support as all Github actions using Node 12 are now deprecated. (https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)

**Review by dariakp — APPROVED** [review]

**Pull request merged**